content/en/docs/components/pipelines/v2/compile-a-pipeline.md

```diff
@@ -1,104 +1,75 @@
-
-**root**  
-[`root`][root-schema] defines the steps of the outermost (root) pipeline definition. It is itself a [`ComponentSpec`][component-spec]. This is the pipeline executed when the YAML is submitted.
-
-[Example][root-example]
-
-**pipeline_info**  
-[`pipeline_info`][pipeline-info-schema] contains pipeline metadata, including the pipeline name.
-
-[Example][pipeline-info-example]  
-
-**sdk_version**  
-[`sdk_version`][sdk-version-schema] records which version of the KFP SDK compiled the pipeline.
-
-[Example][sdk-version-example]  
 
-**schema_version**  
-[`schema_version`][schema-version-schema] records which version of the `PipelineSpec` schema is used for the IR YAML.
+The IR YAML is an intermediate representation of a compiled pipeline or component. It is an instance of the [`PipelineSpec`][pipeline-spec] protocol buffer message type, which is a platform-agnostic pipeline representation protocol. It is considered an intermediate representation because the KFP backend compiles `PipelineSpec` to [Argo Workflow][argo-workflow] YAML as the final execution definition for execution on Kubernetes.
 
-[Example][schema-version-example]
+Unlike the v1 component YAML, the IR YAML is not intended to be written directly. To learn how to author pipelines and components in KFP v2 similar to authoring component YAML in KFP v1, see [Author a Pipeline: Custom Container Components][custom-container-component-authoring].
 
-**default_pipeline_root**  
-[`default_pipeline_root`][default-pipeline-root-schema] is the remote storage root path where the pipeline outputs will be written, such as a Google Cloud Storage URI (`gcs://my/path`).
+The compiled IR YAML file contains the following sections:
 
-[Example][default-pipeline-root-example]  
+| Section | Description | Example |
+|-------|-------------|---------|
+| [`components`][components-schema] | This section is a map of the names of all components used in the pipeline to [`ComponentSpec`][component-spec]. `ComponentSpec` defines the interface, including inputs and outputs, of a component.<br/>For primitive components, `ComponentSpec` contains a reference to the executor containing the component implementation.<br/>For pipelines used as components, `ComponentSpec` contains a [DagSpec][dag-spec] instance, which includes references to the underlying primitive components. | [View on Github][components-example]
+| [`deployment_spec`][deployment-spec-schema] | This section contains a map of executor name to [`ExecutorSpec`][executor-spec]. `ExecutorSpec` contains the implementation for a primitive component. | [View on Github][deployment-spec-example]
+| [`root`][root-schema] | This section defines the steps of the outermost pipeline definition, also called the pipeline root definition. The root definition is an instance of [`ComponentSpec`][component-spec], which is also the pipeline executed on Kubernetes when you submit the IR YAML definition to Kubernetes. | [View on Github][root-example]
```

Comment:
```suggestion
| [`root`][root-schema] | This section defines the steps of the outermost pipeline definition, also called the pipeline root definition. The root definition is the workflow executed when you submit the IR YAML. It is an instance of [`ComponentSpec`][component-spec].
```

There's probably a better way to phrase this than what I've suggested, but the main suggestion is to remove references to Kubernetes or submitting IR YAML to Kubernetes.

The IR YAML gets submitted to the KFP BE (then the KFP BE submits Argo Workflow to Kubernetes) and IR YAML can be run on platforms that do not use Kubernetes.

Reply:
Following up on this comment.

Reply:
Done.